### PR TITLE
Restructure & New Additions

### PR DESCRIPTION
**Pull request by jhaag75**
Updates:

- Restructured repository root and moved misc files to folders so profile spec files will be easier to locate.
- Added new ontology files from @fugu13 
- Added a few more profiles

TODO: 
- move scorm context and schema files to SCORM profile repo. 
- Update W3ID identifiers to point to new profile ontology in this repo.

### CODE REVIEW

**Comment by jhaag75** [comment]
Also TODO:
- Update persistent W3ID profile context IRI to point to the profile-context.jsonld in this repo
- update @context IRIs in the profile examples to point to profile context IRI

File: profiles/adb.jsonld [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  ],
+  "concepts": [
+    {
+      "id": "https://w3id.org/xapi/adb/verbs/annotated",
+      "inScheme": "https://w3id.org/xapi/adb",
+      "type": "Verb",
+      "definition": {
+        "en": "Provide notes or symbols for further explanation or thoughts while reading."
+      },
+      "narrower": [ {
```
Comment (fugu13) [inline profiles/adb.jsonld:32]: This is not the correct syntax for narrower. Narrower's value is an array of IRIs, not an array of objects.

File: profiles/adb.jsonld [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+          "id": "https://w3id.org/xapi/adb/verbs/highlighted"
+      } ],
+      "relatedMatch": [ {
```
Comment (fugu13) [inline profiles/adb.jsonld:35]: Must be array of IRIs

File: profiles/adb.jsonld [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    {
+      "id": "https://w3id.org/xapi/adb/v1",
+      "generatedAtTime": "2017-06-30T8:26:00Z",
+      "wasRevisionOf": ["https://w3id.org/xapi/adb"]
+    }
+  ],
+  "concepts": [
+    {
+      "id": "https://w3id.org/xapi/adb/verbs/annotated",
+      "inScheme": "https://w3id.org/xapi/adb",
```
Comment (fugu13) [inline profiles/adb.jsonld:27]: inScheme needs to be of the version; it's already contained in the profile overall via the concepts relation

File: profiles/adb.jsonld [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+      "type": "Verb",
+      "definition": {
+        "en": "Provide notes or symbols for further explanation or thoughts while reading."
```
Comment (fugu13) [inline profiles/adb.jsonld:30]: This definition needs to be expanded to guide usage. I'm an expert at xAPI and I don't understand what I should generally expect statements using this verb to look like.

File: profiles/adb.jsonld [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+      } ],
+      	"prefLabel": {
+        "en": "annotated"
+      }
+    },
+    {
+      "id": "https://w3id.org/xapi/adb/verbs/arrived",
+      "inScheme": "https://w3id.org/xapi/adb",
+      "type": "Verb",
+      "broadMatch": {
```
Comment (fugu13) [inline profiles/adb.jsonld:46]: Must be array of IRIs

File: profiles/adb.jsonld [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+        "id": "http://adlnet.gov/expapi/verbs/progressed"
+      },
+      "definition": {
+        "en": "Indicates the actor arrived a specific location by movement or progress. "
+      },
+      "scopeNote": {
```
Comment (fugu13) [inline profiles/adb.jsonld:52]: scopeNote is not legal here

File: profiles/adb.jsonld [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+        "en": "To arrive a specific scene within a story or book experience."
```
Comment (fugu13) [inline profiles/adb.jsonld:53]: Also, the contents of scopeNote as used aren't about scope, they're about what the verb means, which should go in definition anyways

File: profiles/adb.jsonld [hunk 8]
```diff
@@ -0,0 +1,231 @@
...
+      "id": "https://w3id.org/xapi/adb/verbs/coached",
+      "inScheme": "https://w3id.org/xapi/adb",
+      "type": "Verb",
+      "definition": {
+        "en": "To teach or lead another person through an activity. Coached is a direct response to the Requested verb. The receiver of a Requested statement will periodically respond by sending a Coached statement that clears out the coach's queue."
+      },
+        "prefLabel": {
+        "en": "coached"
+      },
+      "related": [ {
```
Comment (fugu13) [inline profiles/adb.jsonld:94]: These are not related. Related only means the meaning is close, it is not a general "there is some sort of relationship". That one is a response to the other, a causal relationship, does not mean their meanings are close -- it means their meanings are very different.

File: profiles/adb.jsonld [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+        "id": "https://w3id.org/xapi/adb/verbs/requested"
+      } ]
+    },
+    {
+      "id": "https://w3id.org/xapi/adb/verbs/demanded",
+      "inScheme": "https://w3id.org/xapi/adb",
+      "type": "Verb",
+      "definition": {
+        "en": "Indicates a request for eBook session or redirect. Shall be distinct from requested which refers to actor conversations (e.g., requested help fom instructor)."
```
Comment (fugu13) [inline profiles/adb.jsonld:103]: What's "a request for eBook session or redirect"?

File: profiles/seriousgames.jsonld [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+        "en": "An object with attributes x, y and z indicating the position of the player within a particular area in a game. Used to represent the current position of the player inside the game world. This position should be expressed in a coordinate system known by the data analysts. This extension appears in the Statement's result."
+      },
+      "prefLabel": {
+        "en": "position"
+      }
+    },
+    {
+      "id": "https://w3id.org/xapi/seriousgames/verbs/accessed",
+      "type": "Verb",
+      "exactMatch": {
```
Comment (fugu13) [inline profiles/seriousgames.jsonld:219]: Must be an array of IRIs

File: profiles/video.jsonld [hunk 11]
```diff
@@ -0,0 +1,551 @@
...
+      "id": "https://w3id.org/xapi/video/v1",
+      "generatedAtTime": "2017-06-29T10:45:00Z",
+      "wasRevisionOf": ["https://w3id.org/xapi/video"]
+      }],
+  "concepts": [
+    {
+      "id": "https://w3id.org/xapi/video/verbs/paused",
+      "inScheme": "https://w3id.org/xapi/video",
+      "type": "Verb",
+      "exactMatch": [ {
```
Comment (fugu13) [inline profiles/video.jsonld:28]: Must be an array of IRIs

File: profiles/video.jsonld [hunk 12]
```diff
@@ -0,0 +1,551 @@
...
+      ]
+    },
+    {
+      "id": "https://w3id.org/xapi/video/templates#basic",
+      "name": {
+        "en": "Video Interactions"
+      },
+      "objectActivityType": "https://w3id.org/xapi/video/activity-type/video",
+      "rules": [{
+        "location": "verb.id",
```
Comment (fugu13) [inline profiles/video.jsonld:498]: Needs to start with `$.`

File: profiles/video.jsonld [hunk 13]
```diff
@@ -0,0 +1,551 @@
...
+          "https://w3id.org/xapi/video/verbs/played",
+          "https://w3id.org/xapi/video/verbs/paused",
+          "https://w3id.org/xapi/video/verbs/seeked"
+        ]
+        }]
+      }
+  ],
+  "patterns": [
+      {
+         "id": "_:generalpattern",
```
Comment (fugu13) [inline profiles/video.jsonld:509]: These should use full URIs

File: profiles/video.jsonld [hunk 14]
```diff
@@ -0,0 +1,551 @@
...
+        }
+      ]
+    },
+    {
+      "id": "https://w3id.org/xapi/video/templates#basic",
+      "name": {
+        "en": "Video Interactions"
+      },
+      "objectActivityType": "https://w3id.org/xapi/video/activity-type/video",
+      "rules": [{
```
Comment (fugu13) [inline profiles/video.jsonld:497]: This doesn't mean what's intended. What this template currently means is, every statement with an object activity type of video can only have one of those three verbs (and since there are other templates above that contradict this, I know it isn't what's intended).

Rules are for restricting the behavior of statements that match the determining properties.

File: profiles/video.jsonld [hunk 15]
```diff
@@ -0,0 +1,551 @@
...
+          "alternates": ["https://w3id.org/xapi/video/templates#generalrestrictions", "https://w3id.org/xapi/video/templates#played", "https://w3id.org/xapi/video/templates#paused", "https://w3id.org/xapi/video/templates#seeked", "https://w3id.org/xapi/video/templates#volumechange", "https://w3id.org/xapi/video/templates#completed"]
+      },
+      {
+          "@id": "_:all-activities-pattern",
+          "alternates": ["https://w3id.org/xapi/video/templates#generalrestrictions", "https://w3id.org/xapi/video/templates#played", "https://w3id.org/xapi/video/templates#paused", "https://w3id.org/xapi/video/templates#seeked", "https://w3id.org/xapi/video/templates#volumechange", "https://w3id.org/xapi/video/templates#screenchange", "https://w3id.org/xapi/video/templates#completed"]
+      },
+      {
+          "id": "_:optionalmiddlestatements",
+          "type": "Pattern",
+			    "zeroOrMore": ["_:played-pattern", "_:played-completed-pattern", "_:played-paused-completed-pattern", "_:played-paused-seeked-completed-pattern", "_:played-paused-seeked-volumechange-completed-pattern", "_:all-activities-pattern"],
```
Comment (fugu13) [inline profiles/video.jsonld:546]: You can't have both zeroOrMore and oneOrMore in the same pattern

Comment (fugu13) [reply profiles/video.jsonld:546]: Also, the value of zeroOrMore is a single identifier, not an array.

File: profiles/video.jsonld [hunk 16]
```diff
@@ -0,0 +1,551 @@
...
+          "oneOrMore" : ["https://w3id.org/xapi/video/templates#basic"]
```
Comment (fugu13) [inline profiles/video.jsonld:547]: Likewise, the value of oneOrMore is a single identifier, not an array

File: profiles/video.jsonld [hunk 17]
```diff
@@ -0,0 +1,551 @@
...
+          "@id": "_:played-paused-seeked-completed-pattern",
+          "alternates": ["_:generalrestrictions", "https://w3id.org/xapi/video/templates#played", "https://w3id.org/xapi/video/templates#paused", "https://w3id.org/xapi/video/templates#seeked", "https://w3id.org/xapi/video/templates#completed"]
+      },
+      {
+          "@id": "_:played-paused-seeked-volumechange-completed-pattern",
+          "alternates": ["https://w3id.org/xapi/video/templates#generalrestrictions", "https://w3id.org/xapi/video/templates#played", "https://w3id.org/xapi/video/templates#paused", "https://w3id.org/xapi/video/templates#seeked", "https://w3id.org/xapi/video/templates#volumechange", "https://w3id.org/xapi/video/templates#completed"]
+      },
+      {
+          "@id": "_:all-activities-pattern",
+          "alternates": ["https://w3id.org/xapi/video/templates#generalrestrictions", "https://w3id.org/xapi/video/templates#played", "https://w3id.org/xapi/video/templates#paused", "https://w3id.org/xapi/video/templates#seeked", "https://w3id.org/xapi/video/templates#volumechange", "https://w3id.org/xapi/video/templates#screenchange", "https://w3id.org/xapi/video/templates#completed"]
```
Comment (fugu13) [inline profiles/video.jsonld:541]: You probably don't want general restrictions here (or anywhere else). Since it matches literally every statement using this profile, if it is used it is superfluous to list anything else in an alternates -- they might as well not be there, and all guidance from a pattern is lost.

File: profiles/video.jsonld (hunk 15 above)
Comment (jhaag75) [reply profiles/video.jsonld:546]: can you have multiple "zeroOrMore" properties, like this? 

`      {
          "id": "https://w3id.org/xapi/video/patterns#optionalmiddlestatements",
          "type": "Pattern",
			    "zeroOrMore": ["https://w3id.org/xapi/video/patterns#played-pattern"], 
          "zeroOrMore": ["https://w3id.org/xapi/video/patterns#played-completed-pattern"], 
          "zeroOrMore": ["https://w3id.org/xapi/video/patterns#played-paused-completed-pattern"], 
          "zeroOrMore": ["https://w3id.org/xapi/video/patterns#played-paused-seeked-completed-pattern"], 
          "zeroOrMore": ["https://w3id.org/xapi/video/patterns#played-paused-seeked-volumechange-completed-pattern"], 
          "zeroOrMore": ["https://w3id.org/xapi/video/patterns#all-activities-pattern"]

      }`

File: profiles/adb.jsonld (hunk 9 above)
Comment (jhaag75) [reply profiles/adb.jsonld:103]: Not sure. Will ask them for a better description.

File: profiles/adb.jsonld (hunk 7 above)
Comment (jhaag75) [reply profiles/adb.jsonld:53]: This was left-over accidentally...will move it to description.

File: profiles/adb.jsonld (hunk 4 above)
Comment (jhaag75) [reply profiles/adb.jsonld:30]: Will ask for a better definition here.

File: profiles/video.jsonld (hunk 15 above)
Comment (fugu13) [reply profiles/video.jsonld:546]: No, you can only have a property once, and its value has to be a string. What are you trying to express with the multiple zeroOrMores? If just that it can be zeroOrMore of any of them, you can restructure like this (I'm using non-URIs for shorthand; they'd all be URIs, of course, but everything else the same):

```javascript
{
   "id": "allthepossibilities",
    "alternates": [
        "https://w3id.org/xapi/video/patterns#played-pattern",
        "another",
        "third"
    ]
},
{
    "id": "https://w3id.org/xapi/video/patterns#optionalmiddlestatements",
    "zeroOrMore": "allthepossibilities"
}
```
